lib/routes/account.js

```diff
@@ -1043,16 +1128,18 @@ module.exports = function (
         log.begin('Account.RecoveryEmailResend', request)
         var sessionToken = request.auth.credentials
         var service = request.payload.service || request.query.service
-        if (sessionToken.emailVerified ||
-            Date.now() - sessionToken.verifierSetAt < config.smtp.resendBlackoutPeriod) {
-          return reply({})
-        }
-        customs.check(
-          request.app.clientAddress,
-          sessionToken.email,
-          'recoveryEmailResendCode')
-          .then(
-            mailer.sendVerifyCode.bind(
+
+        if (!sessionToken.emailVerified) {
+          // Send email verification
+          if (Date.now() - sessionToken.verifierSetAt < config.smtp.resendBlackoutPeriod) {
+            return reply({})
+          }
+
+          return customs.check(
+            request.app.clientAddress,
+            sessionToken.email,
+            'recoveryEmailResendCode')
+            .then(mailer.sendVerifyCode.bind(
               mailer,
               sessionToken,
               sessionToken.emailCode,
```

Comment:
@rfk I think the case you describe is accurate. However, I believe on the content-server side, when doing (3), an email is automatically sent when you login. @shane-tomlinson is that correct?

Also, when calling `/recovery_email/verify_code`, any valid `tokenVerificationId` will verify the account email associated [with it too.](https://github.com/mozilla/fxa-auth-server/blob/verify-token-full/lib/routes/account.js#L1253)

With this, I think we can simplify the logic of `/recovery_email/resend_code` to always send a `tokenVerificationId` associated with current session and have it validate sign-in and account?